test/Identity.IntegrationTest/Endpoints/IdentityServerTests.cs

```diff
@@ -0,0 +1,298 @@
+
+        public IdentityServerTests(IdentityApplicationFactory factory)
+        {
+            _factory = factory;
+        }
+
+        [Fact]
+        public async Task WellKnownEndpoint_Success()
+        {
+            var context = await _factory.Server.GetAsync("/.well-known/openid-configuration");
+
+            using var body = await AssertHelper.ResponseIsAsync<JsonDocument>(context);
+            var root = body.RootElement;
+            Assert.Equal(JsonValueKind.Object, root.ValueKind);
+            AssertHelper.AssertJsonProperty(root, "issuer", JsonValueKind.String);
+
+            var scopesSupported = AssertHelper.AssertJsonProperty(root, "scopes_supported", JsonValueKind.Array)
+                .EnumerateArray()
+                .Select(je => je.GetString());
+
+            Assert.Contains("api", scopesSupported);
+            Assert.Contains("api.push", scopesSupported);
+            Assert.Contains("api.licensing", scopesSupported);
+            Assert.Contains("api.organization", scopesSupported);
+            Assert.Contains("internal", scopesSupported);
+            Assert.Contains("offline_access", scopesSupported);
+
+            var claimsSupported = AssertHelper.AssertJsonProperty(root, "claims_supported", JsonValueKind.Array)
+                .EnumerateArray()
+                .Select(je => je.GetString());
+
+            Assert.Contains("name", claimsSupported);
+            Assert.Contains("email", claimsSupported);
+            Assert.Contains("email_verified", claimsSupported);
+            Assert.Contains("sstamp", claimsSupported);
+            Assert.Contains("premium", claimsSupported);
+            Assert.Contains("device", claimsSupported);
+            Assert.Contains("orgowner", claimsSupported);
+            Assert.Contains("orgadmin", claimsSupported);
+            Assert.Contains("orgmanager", claimsSupported);
+            Assert.Contains("orguser", claimsSupported);
+            Assert.Contains("orgcustom", claimsSupported);
+            Assert.Contains("providerprovideradmin", claimsSupported);
+            Assert.Contains("providerserviceuser", claimsSupported);
+            Assert.Contains("sub", claimsSupported);
+
+            var grantTypesSupported = AssertHelper.AssertJsonProperty(root, "grant_types_supported", JsonValueKind.Array)
+                .EnumerateArray()
+                .Select(je => je.GetString());
+
+            Assert.Contains("authorization_code", grantTypesSupported);
+            Assert.Contains("client_credentials", grantTypesSupported);
+            Assert.Contains("refresh_token", grantTypesSupported);
+            Assert.Contains("implicit", grantTypesSupported);
+            Assert.Contains("password", grantTypesSupported);
+            Assert.Contains("urn:ietf:params:oauth:grant-type:device_code", grantTypesSupported);
+
+            // QUESTION: What are other breaking changes from this configuration that we should assert
```

Comment:
I have a split opinion on this, but I believe this might be a scenario where it makes sense to compare against a static json file.

My argument being that we want to be aware if anything on this endpoint changes. But definitely open for other thoughts, as I typically dislike strictly comparing since they require a bit of maintenance.

Reply:
I like it. I will add that.

Reply:
I think it's a good idea. My feeling was that I wanted the test to break on any removals but not to break with any additions but even additions should be made carefully so I have switched it to strictly compare the JSON based on a static file.